src/domain_bridge/domain_bridge.cpp

```diff
@@ -235,6 +235,9 @@ class DomainBridgeImpl
         rclcpp::PublisherOptionsWithAllocator<std::allocator<void>> publisher_options;
         rclcpp::SubscriptionOptionsWithAllocator<std::allocator<void>> subscription_options;
 
+        // Prevent endless looping on a bidirectional bridge
+        subscription_options.ignore_local_publications = true;
```

Comment:
I'm not sure if all rmw implementations where supporting this, which could be a problem.

Reply:
we cannot do much about that issue here though

Reply:
Is there a way to detect if an rmw doesn't support this? Then at least we can guard against runaway message loops.

Reply:
It's not possible to detect that as far as I know.

* fastrtps seems to support it: https://github.com/ros2/rmw_fastrtps/blob/b5f581f2867cdf6e9e69b3966a9cd8313f1a15ce/rmw_fastrtps_shared_cpp/src/rmw_take.cpp#L91
* cyclonedds too https://github.com/ros2/rmw_cyclonedds/blob/ab3d209ad320757ee2a65c6fc4c940ae49d5d6d8/rmw_cyclonedds_cpp/src/rmw_node.cpp#L1813-L1815
* same for connextdds https://github.com/ros2/rmw_connextdds/blob/9d0dec6270766f2249e27d17280071c68b4a06de/rmw_connextdds_common/src/ndds/dds_api_ndds.cpp#L691-L697

So it seems this is not an issue, I might be remembering an old problem.